types/time.go

```diff
@@ -110,10 +110,12 @@ var (
 	// MaxDatetime is the maximum for mysql datetime type.
 	MaxDatetime = FromDate(9999, 12, 31, 23, 59, 59, 999999)
 
+	// min and max timestamp timezone
+	boundTimezone = gotime.Local
 	// MinTimestamp is the minimum for mysql timestamp type.
-	MinTimestamp = FromDate(1970, 1, 1, 0, 0, 1, 0)
+	MinTimestamp, _ = FromDate(1970, 1, 1, 0, 0, 1, 0).convertTimeZone(gotime.UTC, boundTimezone)
```

Comment:
MinTimestamp and maxTimestamp is a number, a constant, without `TimeZone` information.
If you want to compare, you should not change the constant, instead, change the variable to the right TimeZone and compare with it.